src/rapids_singlecell/preprocessing/_normalize.py

```diff
@@ -55,38 +60,32 @@ def normalize_total(
         adata = adata.copy()
     X = _get_obs_rep(adata, layer=layer)
 
-    _check_gpu_X(X)
+    _check_gpu_X(X, allow_dask=True)
 
     if not inplace:
         X = X.copy()
 
     if sparse.isspmatrix_csc(X):
         X = X.tocsr()
     if not target_sum:
-        if sparse.issparse(X):
-            from ._kernels._norm_kernel import _get_sparse_sum_major
+        target_sum = _get_target_sum(X)
 
-            counts_per_cell = cp.zeros(X.shape[0], dtype=X.dtype)
-            sum_kernel = _get_sparse_sum_major(X.dtype)
-            sum_kernel(
-                (X.shape[0],),
-                (64,),
-                (X.indptr, X.data, counts_per_cell, X.shape[0]),
-            )
-        else:
-            counts_per_cell = X.sum(axis=1)
-        target_sum = cp.median(counts_per_cell)
+    X = _normalize_total(X, target_sum)
 
-    if sparse.isspmatrix_csr(X):
-        from ._kernels._norm_kernel import _mul_csr
+    if inplace:
+        _set_obs_rep(adata, X, layer=layer)
 
-        mul_kernel = _mul_csr(X.dtype)
-        mul_kernel(
-            (math.ceil(X.shape[0] / 128),),
-            (128,),
-            (X.indptr, X.data, X.shape[0], int(target_sum)),
-        )
+    if copy:
+        return adata
+    elif not inplace:
+        return X
 
+
+def _normalize_total(X: cp.ndarray, target_sum: int):
+    if isinstance(X, sparse.csr_matrix):
+        return _normalize_total_csr(X, target_sum)
+    elif isinstance(X, DaskArray):
+        return _normalize_total_dask(X, target_sum)
```

Comment:
@flying-sheep just so you're away when reviewing this, this is why we can't use single dispatch anywhere: https://github.com/sphinx-doc/sphinx/issues/10591

Reply:
possible that this is no longer relevant, i didn't look into it too hard